src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -349,6 +350,15 @@ public Snapshot makeSnapshotFromSnapshotRequest(SnapshotRequestModel snapshotReq
         .collect(Collectors.toList());
   }
 
+  public UUID getFirstSourceDatasetIdFromSnapshotId(
+      UUID snapshotId, AuthenticatedUserRequest userRequest) {
+    SnapshotModel snapshotModel = retrieveAvailableSnapshotModel(snapshotId, userRequest);
+    return snapshotModel.getSource().stream()
+        .map(s -> s.getDataset().getId())
+        .findFirst()
+        .orElseThrow(() -> new DatasetNotFoundException("Source dataset for snapshot not found"));
+  }
+
   public List<UUID> getSourceDatasetIdsFromSnapshotId(
```

Comment:
I think once you have `getFirstSourceDatasetIdFromSnapshotId()` this method can be deleted.